File: 1583/digit.c
#include <stdio.h>

int get_n_digits(int n)
{
	int x = 0;

	while(n > 0)
	{
		x = x + 1;
		n = n / 10;
	}

	return x;
}


int get_sum_digits(int n)
{
	int x = 0;

	while(n > 0)
	{
		x = x + (n % 10);
		n = n / 10;
	}

	return x;
}

int main(int argc, char *argv[])
{
	int t, n;
	int i, j, lim, x, y, z, ans;

	scanf("%d", &t);

	for(i=0; i<t; i++)
	{
		scanf("%d", &n);

		x = get_n_digits(n);
		lim = n - 9*x - 1;
		ans = 0;

		for(j=lim; j<=n; j++)
		{
			y = get_sum_digits(j);
			z = j + y;

			if (z == n)
			{
				ans = j;
				break;
			}
		}

		printf("%d\n", ans);
	}

	return 0;
}

File: 1583/digit.py
import sys

def get_n_digits(n):
	x = 0

	while n > 0:
		x = x + 1
		n = n // 10

	return x

def get_sum_digits(n):
	x = 0

	while n > 0:
		x = x + (n % 10)
		n = n // 10

	return x

t = int(input())

for i in range(t):
	n = int(input())
	n_digits = get_n_digits(n)
	
	lim = n - 9 * n_digits - 1

	flag = False
	for j in range(lim, n+1):
		sum_digits = get_sum_digits(j)
		x = j + sum_digits
		if x == n:
			flag = True
			ans = j
			break

	if flag == True:
		print(ans)
	else:
		print(0)
